src/main/java/it/gov/pagopa/pu/sil/service/querypayments/PaaSILChiediStoricoPagamentiService.java

```diff
@@ -0,0 +1,107 @@
+package it.gov.pagopa.pu.sil.service.querypayments;
+
+import it.gov.pagopa.pu.auth.dto.generated.UserInfo;
+import it.gov.pagopa.pu.debtpositions.dto.generated.DebtPositionDTO;
+import it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentStatus;
+import it.gov.pagopa.pu.debtpositions.dto.generated.PersonEntityType;
+import it.gov.pagopa.pu.organization.dto.generated.Organization;
+import it.gov.pagopa.pu.organization.dto.generated.OrganizationStatus;
+import it.gov.pagopa.pu.sil.connector.debtpositions.DebtPositionService;
+import it.gov.pagopa.pu.sil.connector.organization.service.OrganizationService;
+import it.gov.pagopa.pu.sil.enums.SilFaults;
+import it.gov.pagopa.pu.sil.exception.SilFaultException;
+import it.gov.pagopa.pu.sil.mapper.PagatiMapper;
+import it.gov.pagopa.pu.sil.service.AuthorizationService;
+import it.gov.pagopa.pu.sil.service.receipt.ReceiptService;
+import it.gov.pagopa.pu.sil.util.ByteArrayDataSource;
+import it.gov.pagopa.pu.sil.util.ConversionUtils;
+import it.veneto.regione.pagamenti.ente.PaaSILChiediStoricoPagamenti;
+import it.veneto.regione.pagamenti.ente.PaaSILChiediStoricoPagamentiRisposta;
+import it.veneto.regione.pagamenti.ente.PaaSILStoricoPagamenti;
+import jakarta.activation.DataHandler;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.*;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class PaaSILChiediStoricoPagamentiService {
+
+  private final DebtPositionService debtPositionService;
+  private final OrganizationService organizationService;
+  private final ReceiptService receiptService;
+  private final PagatiMapper pagatiMapper;
+
+  public PaaSILChiediStoricoPagamentiRisposta processRequest(
+    PaaSILChiediStoricoPagamenti request,
+    UserInfo userInfo,
+    String accessToken
+  ) {
+    PaaSILChiediStoricoPagamentiRisposta response = new PaaSILChiediStoricoPagamentiRisposta();
+    AuthorizationService.validateAdminRole(request.getCodIpaEnte(), userInfo);
+
+    Long organizationId = AuthorizationService.getOrganizationIdFromUserInfo(userInfo, request.getCodIpaEnte());
+    Organization organization = organizationService.getOrganizationById(organizationId, accessToken)
+      .orElse(null);
+    if (organization == null || !OrganizationStatus.ACTIVE.equals(organization.getStatus())) {
+      throw new SilFaultException(SilFaults.PAA_ENTE_NON_VALIDO, "L'ente non è valido o non è abilitato");
+    }
+
+    OffsetDateTime dateFrom = LocalDate.now().atStartOfDay().atOffset(ZoneId.systemDefault().getRules().getOffset(Instant.now()));
+    if (request.getDataFrom() != null) {
+      dateFrom = request.getDataFrom().toGregorianCalendar().toZonedDateTime().toOffsetDateTime();
+    }
+
+    OffsetDateTime dateTo = LocalDate.now().atTime(LocalTime.MAX).atOffset(ZoneId.systemDefault().getRules().getOffset(Instant.now()));
```

Comment:
you could store the dateFrom default value (currentDateStart) and here just call currentDateStart.plusDays(1)